lib/gollum-lib/committer.rb

```diff
@@ -94,7 +94,7 @@ def add_to_index(dir, name, format, data, allow_same_ext = false)
 
       dir  = '/' if dir.strip.empty?
 
-      fullpath = ::File.join(*[@wiki.page_file_dir, dir, path].compact)
+      fullpath = ::File.join(*[dir, path].compact)
```

Comment:
I think we don't need the `compact` anymore, since both `dir` and `path` shouldn't be nil at this point. So just `::File.join(dir, path)` should suffice.